qiskit/algorithms/quantum_time_evolution/real/implementations/trotterization/trotter_qrte.py

```diff
@@ -0,0 +1,388 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An algorithm to implement a Trotterization real time-evolution."""
+
+from collections import defaultdict
+from itertools import product
+from typing import Union, List, Dict, Optional
+
+from qiskit.algorithms.quantum_time_evolution.real.implementations.trotterization.trotter_ops_validator import (
+    _validate_hamiltonian_form,
+    _validate_input,
+    _is_op_bound,
+)
+from qiskit.algorithms.quantum_time_evolution.real.qrte import Qrte
+from qiskit.circuit import Parameter
+from qiskit.opflow import (
+    OperatorBase,
+    StateFn,
+    Gradient,
+    commutator,
+    SummedOp,
+    PauliSumOp,
+    PauliOp,
+    CircuitOp,
+)
+from qiskit.circuit.library import PauliEvolutionGate
+from qiskit.quantum_info import Pauli
+from qiskit.synthesis import ProductFormula, LieTrotter
+
+
+class TrotterQrte(Qrte):
```

Comment:
My comment from [qrte.py](qiskit/algorithms/quantum_time_evolution/real/qrte.py) also affects this class.